### PR TITLE
Sp 47 refactor

### LINKED ISSUE #73: sp-47 branch refactor [SP-73]
Refactor sp-47 branch to remove duplicates.

### PR DESCRIPTION
**Pull request by mnashat**
merging sp-47 refactor into sp-47 branch

Closes #73

### CODE REVIEW

File: src/components/Profile/Mocks/mockUser.json [hunk 1]
```diff
@@ -4,5 +4,14 @@
     "faculty": "Art & Science",
     "program": "Anthropology",
     "privateProfile": true
+  },
+  {
```
Comment (kikkiii21) [inline src/components/Profile/Mocks/mockUser.json:8]: do we still need mock data?

File: src/components/StudyRoom/Mocks/mockStudyRoom.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+[
```
Comment (kikkiii21) [inline src/components/StudyRoom/Mocks/mockStudyRoom.json:1]: do we still need mock data? can we remove?

File: src/components/StudyRoom/Mocks/mockStudyRoomMessages.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[
```
Comment (kikkiii21) [inline src/components/StudyRoom/Mocks/mockStudyRoomMessages.json:1]: mock data

File: src/components/StudyRoom/Sockets.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+import io from "socket.io-client"
```
Comment (kikkiii21) [inline src/components/StudyRoom/Sockets.js:1]: is this going to be a problem for deployment since the 5000 is hard coded?

**Review by kikkiii21 — APPROVED** [review]
Good job! Most of my comments are concerning the mock data files. Could we remove them or we still need them? Also, in socket.js, there's a path that says 5000. Is this going to create problems for deployment? We changed all the other paths to not include that 5000

File: src/components/StudyRoom/StudyRoom.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+                { /* Drawer right icon */ }
+                <StudyRoomChatCard width='30.5vw' height='7vh' marginTop='2px' topLeftRadius='0px' topRightRadius='0px'
+                                   bottomLeftRadius='0px' bottomRightRadius='10px' content={<div
+                    style={{width: '100%', height: '100%', background: 'none', border: 'none'}}
+                ><BottomDrawer icon={<SettingsIcon style={{color: '#912338', height: '4vh', width: '4vh'}}/>}
+                               title={'settings page'} content={<StudyRoomSettings/>}/>  asda  </div>}/>
+            </div>
+        </React.Fragment>
+    ), [roomData]);
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoom.js:99]: I don't think we want to re render this whole const when room data change. maybe we should split these into smaller const (one for each section). and re render these sections only when specific data (i.e ChatRoom id, messages, etc)  changes

**Comment by guaj** [comment]
It would be a good practice for code legacy to remove all console.log before merging to main.

**Review by guaj — APPROVED** [review]

**Pull request merged**